wolfcrypt/src/ecc.c

```diff
@@ -2793,6 +2797,14 @@ int wc_ecc_shared_secret(ecc_key* private_key, ecc_key* public_key, byte* out,
        return BAD_FUNC_ARG;
    }
 
+#ifdef WOLF_CRYPTO_DEV
+    if (private_key->devId != INVALID_DEVID) {
+        err = wc_CryptoDev_Ecdh(private_key, public_key, out, outlen);
+        if (err != NOT_COMPILED_IN)
+            return err;
```

Comment:
does err need to be reset in the NOT_COMPILED_IN case here as in other spots?

Reply:
These were removed because scan build reported set but not used. If you look down on all 3 ecc cases ret is always set so it doesn’t need reset to 0.